### PR TITLE
Fix describe ephemeral pvc mount by pod

### PR DESCRIPTION
**Pull request by cwdsuzhou**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

`kubectl describe pvc xxx` will missing Mount by pod if the pvc is an ephemeral pvc

#### Does this PR introduce a user-facing change?

```release-note

NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 1]
```diff
@@ -1667,6 +1667,10 @@ func getPodsForPVC(c corev1client.PodInterface, pvcName string, settings Describ
 		for _, volume := range pod.Spec.Volumes {
 			if volume.VolumeSource.PersistentVolumeClaim != nil && volume.VolumeSource.PersistentVolumeClaim.ClaimName == pvcName {
 				pods = append(pods, pod)
+				continue
+			}
+			if volume.Ephemeral != nil && volume.Ephemeral.VolumeClaimTemplate != nil && pod.Name+"-"+volume.Name == pvcName {
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:1672]: Why do we do this generation `pod.Name+"-"+volume.Name`?. Is it always equal to `pvcName` and could you please give the reference of this generation in code base?

Comment (carlory) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:1672]: A better way is to reuse the [VolumeIsForPod](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-helpers/storage/ephemeral/ephemeral.go)

**Comment by eddiezane** [comment]
Can you please provide a before an after of what this fixes?

/remove-lifecycle rotten

**Comment by carlory** [comment]
@cwdsuzhou Are you still working on it? Can you add a test unit to capture your changes?

**Comment by cwdsuzhou** [comment]
> Can you please provide a before and after of what this fixes?
> 
> /remove-lifecycle rotten

Describe pod does not include ephemeral pvc mount

**Comment by cwdsuzhou** [comment]
> @cwdsuzhou Are you still working on it? Can you add a test unit to capture your changes?

will do.

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 2]
```diff
@@ -1669,6 +1670,10 @@ func getPodsForPVC(c corev1client.PodInterface, pvc *corev1.PersistentVolumeClai
 		for _, volume := range pod.Spec.Volumes {
 			if volume.VolumeSource.PersistentVolumeClaim != nil && volume.VolumeSource.PersistentVolumeClaim.ClaimName == pvc.Name {
 				pods = append(pods, pod)
+				continue
+			}
+			if volume.Ephemeral != nil && volume.Ephemeral.VolumeClaimTemplate != nil && ephemeral.VolumeClaimName(&pod, &volume) == pvcName {
```
Comment (carlory) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:1675]: please use `VolumeIsForPod` instead.

Comment (cwdsuzhou) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go]: done

**Pull request closed without merging**